### DISCUSSION TITLE
Can't use << suppress a field >>

### DISCUSSION THREAD

**Discussion by adomps** [sqlitebrowser/sqlitebrowser#3010] [Q&A]
Hello
I a using DB Browser for SQLite and trying to suppress a column in a table.
After selecting the table, I click on << Modify the table >>, but the option << suppress
a field >> appears in a grey color, meaning that I cannot use it. I have edited the pragmas
to set foreign keys = '0', but this doesn't solve the problem. 
How can I solve this issue ?
Thank you very much.

**Comment by chrisjlocke** [comment]
Can you provide a screenshot?
I'm not sure I know what option you mean.  
When you say 'suppress a field' do you mean delete it, or not display it (hide it) in the 'browse data' tab?
Rather than what isn't working, it can be helpful to know what your end goal is.

**Comment by adomps** [comment]
Here is the screenshot (in french). 
I am a physicist with very little experience in database and I deeply
appreciate receiving your help.


DB Browser Version 3.11.2 (under Ubuntu 20.04)
Built for x86_64-little_endian-lp64, running on x86_64
Version de SQLite 3.31.1
![screenshot_DB_Browser](https://user-images.githubusercontent.com/101818431/162808242-3d7f289f-5175-4e56-aa2e-8ea6934fa8cc.png)

**Reply by adomps** [reply]
I mean << delete a field >> (delete a column), not just hide it.

**Reply by chrisjlocke** [reply]
Aah, that screenshot has helped.  Thanks.

The button removes a field, so you have to select a field first.  As nothing is selected, the button is not active.  Here is a screencast of it in action.

https://user-images.githubusercontent.com/7872218/162829847-863c350c-af34-40ab-95cb-e9ae44c8af76.mp4

If you need a quick breakdown of databases and how DB4S can help, I'd be pleased to help you.

**Comment by adomps** [comment]
This works perfectly ! 
I was looking for some mysterious or subtil reason that would hinder the modification of my database,
but this was just my clumsiness.
Let me thank you warmly once again.
André

**Reply by chrisjlocke** [reply]
No problem.  Glad you got it working in the end. :)  Happy to help.

**Comment by adomps** [comment]
Hello
I am facing a new problem. I want to modify the name of a field in a table, but this is forbidden because this field is referenced as a foreign
key in another table. Therefore, I must first modify this other table and declare that I don't want this foreign key any more. But  I can't achieve this suppression. As you can see in the screenshot, I correctly access to the Foreign Key menu, but I can not proceed forward.
The << reinitialise >> buttons does nothing.
Thanks a lot if you can troubleshoot once again.
![screenshot_DBBrowser](https://user-images.githubusercontent.com/101818431/163412241-5097d8fd-92a7-44dd-bd13-7f163303547b.png)

**Reply by chrisjlocke** [reply]
The short answer is that you can click 'Reset' (or however its translated in your language) which will remove it.

https://user-images.githubusercontent.com/7872218/163436050-cbf73442-9853-494f-b68b-4d2f2e5bb397.mp4

**Reply by chrisjlocke** [reply]
Putting is back is a bit tricky, so this is the full video from where the clip above is taken.
It may not be obvious how the GUI works...

https://user-images.githubusercontent.com/7872218/163436673-75bbc4c0-d490-4b08-9d09-e0e4d46c41db.mp4

Yell if you need detailed instriuctions.

**Comment by adomps** [comment]
Hi 
Thanks for the vide.
Strangely, in the window << Edit Table def >>, I don't have the << Constraints >> tab. When I try to modify a foreign key in the << Field >>
menu,  I get the error message : << database table is locked(DROP TABLE "main.employe";)

**Reply by adomps** [reply]
I finally managed to reset the foreign key that was in trouble. To be sincere, I didn't understand exactly how this miracle happened.
Many thanks for your help.

**Comment by adomps** [comment]
... In fact, this doesn't work. I cannot save the modified database and get the new error message << database is locked (RELEASE "db4s-edittable 1649959895217";)

**Reply by adomps** [reply]
Finally, I had to save the whole database in a new backup file, and working on the backup unlocked the database.
Seems to be fine now, although the underlying behavior remains mysterious to me !

**Reply by chrisjlocke** [reply]
> database is locked (RELEASE "db4s-edittable 1649959895217

This can occur if you edit the database outside of DB4S, or the database becomes locked during editing (again, another DB4S instance, or program editing the database).
Also, if it can't change/edit the journal file.
